parlai/core/torch_agent.py

```diff
@@ -844,6 +876,10 @@ def _warmup_lr(step):
         else:
             self.warmup_scheduler = None
 
+        self._lr_period = self.opt.get('lr_period_updates')
+        if self._lr_period is None or self._lr_period <= 0:
+            self._lr_period = self.opt.get('warmup_updates', -1)
```

Comment:
why do we set it to warmup updates here?

Reply:
This is legacy behavior. Currently `--lr-scheduler` uses `warmup_updates` to set the decay rate. This lets the user have separate those values, but still defaults to the old behavior of depending on `--warmup-updates`. (Previously it was actually not possible to use invsqrt without warmup)